src/Microsoft.HttpRepl.IntegrationTests/Commands/EchoCommandTests.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.HttpRepl.IntegrationTests.SampleApi;
+using Xunit;
+
+namespace Microsoft.HttpRepl.IntegrationTests.Commands
+{
+    public class EchoCommandTests : BaseIntegrationTest, IClassFixture<HttpCommandsFixture<SampleApiServerConfig>>
+    {
+        private readonly SampleApiServerConfig _serverConfig;
+
+        public EchoCommandTests(HttpCommandsFixture<SampleApiServerConfig> fixture)
+        {
+            _serverConfig = fixture.Config;
+        }
+
+        [Fact]
+        public async Task WithEchoOn_ShowsCorrectOutput()
```

Comment:
Should these also do something to verify the echo, testing that the setting is working, not just that the message is written when being toggled?

Reply:
I guess it's covered elsewhere, in the SetHeaderCommandTests below.

Reply:
Closing this. Let me know if we need anything here. Thanks!